core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+public class PartitionStats implements StructLike {
+
+  private static final int STATS_COUNT = 12;
+
+  private PartitionData partition;
```

Comment:
Shall we use `StructLike` given that we no longer need to use this as key in `ConcurrentHashMap`. I think the computation should rely on `StructLikeMap` now. Then we won't need changes in `PartitionUtil` and the implementation would be generic.

Reply:
I tried this and my testcases started failing. 

Maybe because I didn't  implement `StructLikeWrapper` based equals and hashcode. Will explore further.

Reply:
Reason is `StructLike` partitionData cannot be the key for the map. It screws up comparision. 

So, During map population I have used `PartitionData` and the `PartitionStats` maintains  `StructLike` member.